maze-solving-mouse/sigopt_maze_analysis.py

```diff
@@ -0,0 +1,535 @@
+        :param verbose: A function to call to receive the current iteration and determine if progress should be output
+        :type verbose: callable or bool (if bool and True will be output after each tenth test)
+        :param halton_kwargs: kwargs to pass to the generate_halton_points function
+        :type halton_kwargs: dictionary (defaults to {'shift': numpy.random.random()})
+        :return: Arrays recording the mean, standard deviation results, and the left, up and right values tested
+        :rtype: 5 numpy.array objects
+        """
+        num_tests = num_tests or self.num_tests
+        verbose = (lambda it: it % 10 == 0) if verbose is True else verbose
+        assert verbose in (False, None) or callable(verbose)
+        halton_kwargs = {} if halton_kwargs is None else {'shift': numpy.random.random()}
+        if log_sample:
+            test_points = numpy.exp(generate_halton_points(num_points, numpy.log(self.domain), **halton_kwargs))
+        else:
+            test_points = generate_halton_points(num_points, self.domain, **halton_kwargs)
+
+        mean_vec = numpy.empty(num_points)
+        sd_vec = numpy.empty(num_points)
+        start = default_timer()
+        for k, test_point in enumerate(test_points):
+            mean_vec[k], sd_vec[k] = self._execute_assignments(num_tests, *test_point)
+            if verbose and verbose(k):
+                end = default_timer()
+                print '{0:5d} iterations completed, {1:4.1f} seconds since last report'.format(k, end - start)
+                start = end
+
+        left_vec, up_vec, right_vec = test_points.T
+        return mean_vec, sd_vec, left_vec, up_vec, right_vec
+
+    def create_sigopt_experiment(self, name=None):
+        self.experiment = self.conn.experiments().create(
+            name=name or 'Scalarized Mouse Maze Solver',
+            parameters=[
+                {'name': 'left_prob_log', 'bounds': {'max': numpy.log(100), 'min': numpy.log(0.01)}, 'type': 'double'},
+                {'name': 'up_prob_log', 'bounds': {'max': numpy.log(100), 'min': numpy.log(0.01)}, 'type': 'double'},
+                {'name': 'right_prob_log', 'bounds': {'max': numpy.log(100), 'min': numpy.log(0.01)}, 'type': 'double'},
+            ],
+        )
+
+    def _extract_data_from_experiment(self, experiment_id=None):
+        experiment_id = experiment_id or self.experiment.id
+        num_observations = self.conn.experiments(experiment_id).fetch().progress.observation_count
+        mean_vec = numpy.empty(num_observations)
+        sd_vec = numpy.empty(num_observations)
+        right_vec = numpy.empty(num_observations)
+        up_vec = numpy.empty(num_observations)
+        left_vec = numpy.empty(num_observations)
+
+        for k, observation in enumerate(self.conn.experiments(experiment_id).observations().fetch().iterate_pages()):
+            mean_vec[k] = observation.metadata['mean']
+            sd_vec[k] = observation.metadata['std_dev']
```

Comment:
The value that SigOpt is optimizing is not actually the mean or standard deviation, it's some scalarized (combined) version of them.  They are reported as metadata so I can keep track of the components that formed the actual scalar objective that SigOpt is optimizing.

The reason I spent a little extra time on this is because this is how I envision a customer of ours doing multicriteria stuff given our current setting, so I wanted to have an example we could show that looks clean without too much difficulty.  This is also a prototype, to some degree, of how we can implement multicriteria stuff internally.